tests/conftest.py

```diff
@@ -15,6 +17,10 @@
 # Early diagnostic for failed imports
 import pybind11_tests
 
+if os.name != "nt":
```

Comment:
nit Can you add a comment as to why we're doing this here?
`git blame` is decent for more context, but I think two small sentences can help explain, e.g.:

```
# We want to shy away from Python's default choice for Linux of "fork", as it can incur
# post-fork resource contention issues (such as thread ids, etc.).
```

I'm vaguely familiar with this, and with our codebase we hit it with either sledgehammers of:
- "yes fork, but pin down threading for fast-but-reckless multiprocessing spawns"
- "no fork, please use spawn, and don't think about it" (I still need to read about `forkserver` and how it may interact with imports like `numpy` or `cv2` that may configure / spawn threads at import)
https://github.com/EricCousineau-TRI/repro/commit/2cff386a1c0dd6e19c0fe3de4d314a62f7dcbe76

Reply:
> Can you add a comment as to why we're doing this here?

Done. I tried to keep it concise.

Reply:
Hm... While this and the issue describes the symptoms, is there any way you can link to something that provides readers more context about the root cause of why threads + fork == bad / flaky?
(feel free to dismiss if you wanna add to issue discussion itself, or punt)